### PR TITLE
Layer 2: do not listen on interfaces with NOARP flag

### LINKED ISSUE #351: Layer 2: Metallb shouldn't reply on interface with NOARP flag
**Is this a bug report or a feature request?**:

Feature request

**What happened**:

Speaker sends replies to any interface, even if arp disabled for it 

**What you expected to happen**:

If interface have NOARP flag set, speaker should ignore it.

**How to reproduce it (as minimally and precisely as possible)**:

```
ip link set dev <interface> arp off
```
Then try to arping

**Anything else we need to know?**:

Related #349

**Environment**:
- MetalLB version:
- Kubernetes version:
- BGP router type/version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):

### PR DESCRIPTION
**Pull request by kvaps**
**What this PR does / why we need it**:

This PR disables listening on interfaces which have NOARP flag set.

**Which issue(s) this PR fixes** :

Fixes #351

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by kvaps** [comment]
Do not merge, not completed yet

File: internal/layer2/announcer.go [hunk 1]
```diff
@@ -67,6 +68,10 @@ func (a *Announce) updateInterfaces() {
 		if _, err := os.Stat("/sys/class/net/" + ifi.Name + "/master"); !os.IsNotExist(err) {
 			continue
 		}
+		// Check for NOARP flag
+		if b, _ := ioutil.ReadFile("/sys/class/net/" + ifi.Name + "/flags"); b[4:5] >= 56 {
```
[the remark this answers is not in the document]
Comment (kvaps) [reply internal/layer2/announcer.go]: The last change taking 3rd byte from the end, and comparing it.

It is can only have value `[0-f]` or `x`, then I have condition: 

* `b >= 56` - more or equal `"8"`
* `b <= 102` - less or equal than `"f"`

I don't think that we can find something more elegant without adding additional modules

[the remark this answers is not in the document]
Comment (kvaps) [reply internal/layer2/announcer.go]: Yep, we can load `strconv.ParseInt` but what condition you want to use for this number later?

**Comment by kvaps** [comment]
@danderson, please review now

**Comment by kvaps** [comment]
Please review

**Comment by danderson** [comment]
Merged, thanks!

**Pull request merged**